test/main.js

```diff
@@ -305,4 +310,89 @@ describe('gulp-istanbul', function () {
         });
     });
   });
+
+  describe('istanbul.enforceThresholds()', function () {
+    beforeEach(function (done) {
+      // set up coverage
+      gulp.src([ 'test/fixtures/lib/*.js' ])
+        .pipe(istanbul())
+        .pipe(istanbul.hookRequire())
+        .on('finish', done);
+    });
+
+    afterEach(function () {
+      rimraf.sync('coverage');
+      rimraf.sync('cov-foo');
+    });
+
+    it('checks coverage fails against thresholds', function (done) {
```

Comment:
Which threshold are we really checking here? It'd be better to have a test for the global and a test for the each.